.github/workflows/ci-e2e-test.yaml

```diff
@@ -1,15 +1,120 @@
-name: E2E Test
+name: E2E Workflow
 on:
-  workflow_dispatch: {}
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'pkg/**'
+      - 'e2e/**'
+      - 'cmd/**'
+      - 'Dockerfile*'
+      - 'go.mod'
+      - 'go.sum'
+      - '.golangci.yaml'
 
 jobs:
+  init:
+    outputs:
+      tests: ${{ steps.e2es.outputs.result }}
+    runs-on: [ default ]
+    name: "Prepare E2E Scenarios"
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      # find all e2e scenarios in the e2e directory and generate an array of scenario names
+      - name: "E2E Detection"
+        id: e2es
+        run: |
+          scenarios=$(find ${PWD}/e2e -type f -name 'e2e_test.go' -exec dirname {} \; | xargs -n 1 basename | jq -R -s -c 'split("\n")[:-1]')
+          echo $scenarios
+          echo "result=$scenarios" >> $GITHUB_OUTPUT
+
   e2e:
-    runs-on: "default"
+    name: "Run ${{ matrix.e2es }}(${{ matrix.k8s-version}}) E2E"
+    if: needs.init.outputs.tests != '[]'
+    needs: [ init ]
+    runs-on: [ default ]
+    env:
+      ADMIN_CLUSTER: greenhouse-admin
+      REMOTE_CLUSTER: greenhouse-remote
+
     strategy:
       fail-fast: false
       matrix:
-        k8s-version: ["v1.29.9", "v1.30.5", "v1.31.1"]
+        k8s-version: [ "v1.29.8", "v1.30.4", "v1.31.0" ]
+        e2es: ${{fromJson(needs.init.outputs.tests)}}
     steps:
-      - uses: cloudoperators/common/workflows/e2e@main
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      # Create the admin cluster with latest kubernetes version
+      - name: Create Admin Cluster
+        uses: helm/kind-action@v1.10.0
         with:
-          kubernetes-version: ${{ matrix.k8s-version }}
+          version: 'v0.24.0'
```

Comment:
I would not hardcode the used kinD version, rather use what's coming by default by the action developer.
Do you have any reason to use another version and not what is used by the action developer ?
which is currently [v0.22.0](https://github.com/helm/kind-action/blob/c93960c112bd56edb7ac543925e4d5c3ec203dfc/action.yml#L11C15-L11C22)
```suggestion
```

Reply:
I wanted the admin cluster always needs to be on latest version of kubernetes so that performing kubernetes version upgrades in our gardener cluster would be kind of pre-validated.

`KinD` release [v0.22.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.22.0) does not contain the kubernetes v1.30.x / the latest version.

I guess we can pin the node_image to latest version for now and remove the parameter